src/ng/filter/orderBy.js

```diff
@@ -177,7 +177,7 @@ orderByFilter.$inject = ['$parse'];
 function orderByFilter($parse) {
   return function(array, sortPredicate, reverseOrder) {
 
-    if (!(isArrayLike(array))) return array;
+    if (!(isArrayLike(array))) throw minErr('orderBy')('notarray', 'Expected array but received: {0}', array);
```

Comment:
In `filterFilter` we only allow null, undefined and array-like (and I think that should we do here as well). So it would be more like:

```js
if (array == null) return array;
```